### PR TITLE
Python port objodm - 1st Commit of 'objRun.py'

### PR DESCRIPTION
**Pull request by GeospatialDaryl**
Hi-

I finished a clean commit of ojbRun.py.  In this commit:
-  Increased PEP-8 compliance for line length
-  Created the ODMJob class
  - instantiates to a single job, and __init__s a folder of jpg into
  - ODMPhoto class objects, which contain all metadata in object attributes.
-  Factored the first two 'steps'/calls into the ODMJob class as methods
-  Cleaned up some functions and a bit of organization.

The code base is cleaner and easier to read as well.

@dakotabenjamin - thanks for the feedback yesterday on the 'unable to merge' - I was having trouble with my local git install. 

ODM Dev Community - I strongly feel this OO approach is the way to go - it facilitates building a GUI and increases the modularity of individual steps in the processing chain.  I will add some additional comments to try to clarify the class structure - right now, the legacy list properties are kept using the original handles/names, and then, when parsed, added to the usually-more-descriptive object attribute name.

Keep flying them kites, and quads, and hot-air balloons-

Daryl

### CODE REVIEW

**Comment by dakotabenjamin** [comment]
I got an error trying to run jhead: 

```
Traceback (most recent call last):
  File "/odm/objRun.py", line 1228, in <module>
    objODMJob = prepare_objects()
  File "/odm/objRun.py", line 796, in prepare_objects
    objODMJob = ODMJob(CURRENT_DIR, args)
  File "/odm/objRun.py", line 113, in __init__
    self.listObjPhotos.append( ODMPhoto( filename, self) )        
  File "/odm/objRun.py", line 309, in __init__
    srcProcess = subprocess.Popen(cmdSrc, stdout=subprocess.PIPE)
  File "/usr/lib/python2.7/subprocess.py", line 710, in __init__
    errread, errwrite)
  File "/usr/lib/python2.7/subprocess.py", line 1327, in _execute_child
    raise child_exception
OSError: [Errno 2] No such file or directory
```

line 307-9 in objRun.py: 

```
+        #start pipe for jhead
+        cmdSrc = BIN_PATH + os.sep + "jhead "+ CURRENT_DIR + os.sep + self.strFileName
+        srcProcess = subprocess.Popen(cmdSrc, stdout=subprocess.PIPE)
```

**Comment by GeospatialDaryl** [comment]
Thanks- I saw this last night.  The primary purpose for the pull request
was to get the overall structure and object class definitions up for
review.  There is still a good bit to be done on 'objRun.py' to get
production.

Thanks!

Daryl

On Thu, Sep 24, 2015 at 8:30 AM, Dakota Benjamin notifications@github.com
wrote:

> I got an error trying to run jhead:
> 
> Traceback (most recent call last):
>   File "/odm/objRun.py", line 1228, in <module>
>     objODMJob = prepare_objects()
>   File "/odm/objRun.py", line 796, in prepare_objects
>     objODMJob = ODMJob(CURRENT_DIR, args)
>   File "/odm/objRun.py", line 113, in **init**
>     self.listObjPhotos.append( ODMPhoto( filename, self) )
>   File "/odm/objRun.py", line 309, in **init**
>     srcProcess = subprocess.Popen(cmdSrc, stdout=subprocess.PIPE)
>   File "/usr/lib/python2.7/subprocess.py", line 710, in **init**
>     errread, errwrite)
>   File "/usr/lib/python2.7/subprocess.py", line 1327, in _execute_child
>     raise child_exception
> OSError: [Errno 2] No such file or directory
> 
> line 307-9 in objRun.py:
> -        #start pipe for jhead
> -        cmdSrc = BIN_PATH + os.sep + "jhead "+ CURRENT_DIR + os.sep + self.strFileName
> -        srcProcess = subprocess.Popen(cmdSrc, stdout=subprocess.PIPE)
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/OpenDroneMap/OpenDroneMap/pull/175#issuecomment-142964017
> .

## 

---

Daryl Van Dyke
GIS Analyst
Klamath Strategic Habitat Conservation Team
US Fish & Wildlife Service - AFWO, R8
(707) 825-5153
https://github.com/GeospatialDaryl

**Comment by dakotabenjamin** [comment]
As far as structure goes, would it make sense for the resize function to be a method within ODMPhoto?

**Comment by edgarriba** [comment]
@GeospatialDaryl @dakotabenjamin  take a look in here:
https://github.com/edgarriba/OpenDroneMap/tree/cmake

I've created a cmake building system where some parts such bundler are removed from the main pipeline since we are using opensfm as well.

Could be nice to join both branches

File: objRun.py [hunk 1]
```diff
@@ -0,0 +1,1270 @@
...
+    
+                file_resolution = runAndReturn('jhead "' + objPhotos.dictStepVals["step_0_resizedImage"] \
+                                               + '"', 'grep "Resolution"')
+                match = re.search(": ([0-9]*) x ([0-9]*)", file_resolution)
+                if match:
+                    objPhotos.width = int(match.group(1).strip())
+                    objPhotos.height = int(match.group(2).strip())
+                print "\t (" + str(objPhotos.width) + " x " + str(objPhotos.height) + ")"
+    
+    def getKeypoints(self):
```
Comment (edgarriba) [inline objRun.py:149]: this method it's internally implemented in opensfm

File: objRun.py [hunk 2]
```diff
@@ -0,0 +1,1270 @@
...
+                        print "using existing " + self.jobDir + "/" +objPhotos.strFileNameBase + ".key.bin"
+    
+        siftDest = open(self.dictJobLocations["step_1_vlsift"], 'w')
+        siftDest.write(vlsiftJobs)
+        siftDest.close()
+    
+        run("\"" + BIN_PATH + "/parallel\" --no-notice --halt-on-error 1 -j+0 < \"" \
+            + self.dictJobLocations["step_1_vlsift"] + "\"")
+        
+    def match():
```
Comment (edgarriba) [inline objRun.py:202]: this method it's internally implemented in opensfm

**Pull request closed without merging**